### PR TITLE
feat: Add `CelebrityFinder` new algorithm with Junit tests

### PR DESCRIPTION
**Pull request by Hardvan**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

**Comment by alxkm** [comment]
Could you please provide link to some description of this algorithm, because now it looks like some leetcode

**Comment by Hardvan** [comment]
Here are a list of resources that explain the Celebrity Finding algorithm using Stacks:
- [Scaler - The Celebrity Problem](https://www.scaler.in/the-celebrity-problem/)
- [Unraveling the Celebrity Problem: A Solution Using Stacks](https://blogs.brain-mentors.com/unraveling-the-celebrity-problem-a-solution-using-stacks/)

File: src/main/java/com/thealgorithms/stacks/CelebrityFinder.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    /**
+     * Finds the celebrity in the given party matrix.
+     *
+     * @param party A 2D matrix where party[i][j] is 1 if i knows j, otherwise 0.
+     * @return The index of the celebrity, or -1 if there is no celebrity.
+     */
+    public static int findCelebrity(int[][] party) {
+        int n = party.length;
```
Comment (alxkm) [inline src/main/java/com/thealgorithms/stacks/CelebrityFinder.java:22]: Will be better to using directly party.length

File: src/test/java/com/thealgorithms/stacks/CelebrityFinderTest.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package com.thealgorithms.stacks;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+public class CelebrityFinderTest {
+
+    @Test
```
Comment (alxkm) [inline src/test/java/com/thealgorithms/stacks/CelebrityFinderTest.java:9]: Could you please refactor this to parametrized tests approach?

**Review by alxkm — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/stacks/CelebrityFinder.java (hunk 1 above)
Comment (Hardvan) [reply src/main/java/com/thealgorithms/stacks/CelebrityFinder.java:22]: Done!

File: src/test/java/com/thealgorithms/stacks/CelebrityFinderTest.java (hunk 2 above)
Comment (Hardvan) [reply src/test/java/com/thealgorithms/stacks/CelebrityFinderTest.java:9]: Done!

**Review by alxkm — APPROVED** [review]

**Pull request merged**